### PR TITLE
Kyle/launch review handling

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

This PR:
- adds creating a queue for launch add if the default queue does not exist
- adds better error handling to several cases like missing packages and api errors
- fixes the run config overrides being busted
- handles most comments related to CVPs review except for the high level api ones.

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------

------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -948,14 +962,14 @@ def launch(
...
 ):
     """
-    Run an W&B project from the given URI.
-    For local runs, the run will block until it completes.
-    Otherwise, the project will run asynchronously.
-    If running locally (the default), the URI can be either a Git repository URI or a local path.
-    If running on Databricks, the URI must be a Git repository.
-    By default, Git projects run in a new working directory with the given parameters, while
-    local projects run from the project's root directory.
+    Run a W&B run from the given URI. Which can either be a wandb URI or a github repo uri.
```
Comment (stephchen) [inline wandb/cli/cli.py:965]: we also support local directory paths

File: wandb/sdk/internal/internal_api.py [hunk 2]
```diff
@@ -804,18 +804,80 @@ def get_project_run_queues(self, entity, project):
...
+                    return
+                queue_id = res["queueID"]
+
+            else:
+                wandb.termwarn("Unable to push to run queue {}. Queue not found.")
+                return None
         elif len(matching_queues) > 1:
-            logger.error("Multiple queues with name {} found".format(queue_name))
-        queue_id = matching_queues[0]["id"]
+            wandb.termwarn(
```
Comment (stephchen) [inline wandb/sdk/internal/internal_api.py:875]: idk how much this matters but i think termerror here and termlog on line 852 would make more sense?

File: wandb/sdk/launch/_project_spec.py [hunk 3]
```diff
@@ -147,6 +147,9 @@ def _fetch_project_local(self, api: Api) -> None:
                 self.project_dir, run_info["git"]["remote"], run_info["git"]["commit"]
             )
             patch = utils.fetch_project_diff(self.uri, api)
+            if run_info.get("python"):
+                with open(os.path.join(self.project_dir, "runtime.txt"), "w") as fp:
```
Comment (stephchen) [inline wandb/sdk/launch/_project_spec.py:151]: do you want this in aux_dir instead, i thought project_dir was just for code

File: wandb/sdk/launch/docker.py [hunk 4]
```diff
@@ -38,27 +41,33 @@ def validate_docker_installation() -> None:
...
     """
     Uses project and entry point to generate the docker image
     """
-    path = project.project_dir
+    path = launch_project.project_dir
     # this check will always pass since the dir attribute will always be populated
     # by _fetch_project_local
+    get_module(
+        "repo2docker",
+        required='wandb launch requires additional dependencies, install with pip install "wandb[launch]"',
```
Comment (stephchen) [inline wandb/sdk/launch/docker.py:64]: oh this is cool

File: wandb/sdk/launch/docker.py [hunk 5]
```diff
@@ -120,40 +133,58 @@ def build_docker_image(
...
         "ENV WANDB_BASE_URL={base_url}\n"
         "ENV WANDB_API_KEY={api_key}\n"
-        "ENV WANDB_PROJECT={wandb_project}\n"
-        "ENV WANDB_ENTITY={wandb_entity}\n"
-        "ENV WANDB_NAME={wandb_name}\n"
+        "ENV WANDB_PROJECT={project}\n"
+        "ENV WANDB_ENTITY={entity}\n"
         "ENV WANDB_LAUNCH=True\n"
         "ENV WANDB_LAUNCH_CONFIG_PATH={config_path}\n"
         "ENV WANDB_RUN_ID={run_id}\n"
```
Comment (stephchen) [inline wandb/sdk/launch/docker.py:169]: is the change to move this to the run command in a different pr?

File: wandb/sdk/launch/docker.py [hunk 6]
```diff
@@ -232,14 +263,21 @@ def _get_docker_image_uri(name: str, work_dir: str) -> str:
...
+) -> str:
     """
     Creates build context tarfile containing Dockerfile and project code, returning path to tarfile
     """
     directory = tempfile.mkdtemp()
     try:
         dst_path = os.path.join(directory, "wandb-project-contents")
         shutil.copytree(src=work_dir, dst=dst_path)
+        if os.path.exists(config_path):
+            shutil.copyfile(
```
Comment (stephchen) [inline wandb/sdk/launch/docker.py:277]: what is this for just curious, didn't realize we had another intermediate copy going on before the docker image

File: wandb/sdk/launch/launch.py [hunk 7]
```diff
@@ -150,22 +151,22 @@ def run(
...
         docker_args = {}
-    if _is_wandb_local_uri(api.settings("base_url")):
-        if sys.platform == "win32":
-            docker_args["net"] = "host"
-        else:
-            docker_args["network"] = "host"
-        if sys.platform == "linux" or sys.platform == "linux2":
-            docker_args["add-host"] = "host.docker.internal:host-gateway"
+    # if _is_wandb_local_uri(api.settings("base_url")):
+    #     if sys.platform == "win32":
```
Comment (stephchen) [inline wandb/sdk/launch/launch.py:155]: is this supposed to be commented out

File: wandb/sdk/launch/docker.py (hunk 5 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/docker.py:169]: Ah forgot to merge that one in.

File: wandb/sdk/launch/docker.py (hunk 6 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/docker.py:277]: This is the build context path, docker only allows you to copy files from your build context path into the docker container. We could just store the config directly in the LaunchProject, in fact I think thats better. So I'll do that, and just write it to the build context path here

File: wandb/sdk/launch/launch.py (hunk 7 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/launch.py:155]: Ah, needs to be deleted. I wanted to make sure that removing this didn't break tests on circle

File: wandb/sdk/launch/_project_spec.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/_project_spec.py:151]: ah, you're right. I'll follow a similar pattern to the run_config pattern I proposed in another comment.

File: wandb/sdk/internal/internal_api.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/internal/internal_api.py:875]: Yes you're right

**Pull request merged**